### ISSUE TITLE
feature request: parse duration config strings like `1s` into `kotlin.time.Duration` 

### ISSUE THREAD

**Issue by dpozinen** [spring-projects/spring-boot#37562]
With an `application.yml` like so:

```yml
my-app:
  duration: 4h
```
and a class property like so:

```kotlin
import kotlin.time.Duration

@Service
class MyClass(
   @Value("\${my-app.duration}") private val followDuration: Duration
)
```

getting

```log
Unsatisfied dependency expressed through constructor parameter 3: Failed to convert value of type 'java.lang.String' to required type 'long'; For input string: "PT4h"
...
Caused by: org.springframework.beans.TypeMismatchException: Failed to convert value of type 'java.lang.String' to required type 'long'; For input string: "PT4h"
...
Caused by: java.lang.NumberFormatException: For input string: "PT4h"
```

Spring boot version: 3.1.4
Kotlin: 1.9.20-Beta

**Comment by Sap004** [comment]
@dpozinen I think you are facing this issue because

converter provided by Spring Boot works with java.time.Duration and not kotlin.time.Duration

**Comment by dpozinen** [comment]
@Sap004 yes I assume that is the case as well, hence the feature request

**Comment by Sap004** [comment]
@dpozinen I am currently working on addressing this issue. I'll be submitting a pull request shortly for review

**Comment by wilkinsona** [comment]
I'd like to keep this as waiting for triage until we know what's necessary. If it requires further Kotlin code we'll need to consider if the potential benefits justify that complexity. There's also the Framework side of things to consider such as https://github.com/spring-projects/spring-framework/pull/30396.

**Comment by wilkinsona** [comment]
@Sap004 please be aware that, unfortunately, there's a good chance that a PR will be declined at this time. Things are moving towards duration parsing being handling in Spring Framework and https://github.com/spring-projects/spring-framework/pull/30396 is tracking that and we may not want to add any more Boot-specific features in this area until that PR's been resolved one way or another.

**Comment by neeme-praks-sympower** [comment]
> Since Kotlin has an extension method to convert a `java.time.Duration` to a `kotlin.time.Duration` we recommend that the Java type is used for binding and extension method be used whenever a Kotlin type is needed.

The best thing about `kotlin.time.Duration` is the support for human readable period definitions such as `1.5s`. Compare that to `PT1.5S`. More readable format is especially important in case of configuration files (the main use-case for me). Couldn't you just detect that if the target type is `kotlin.time.Duration` then parse it as the Kotlin type?

**Comment by simonbasle** [comment]
I plan on getting back on the [Spring Framework side of things](https://github.com/spring-projects/spring-framework/pull/30396#issuecomment-2037854277) soon. Regarding the Kotlin-style parsing, I fear this might be a bit too involved to fully support decimal parts.

I also wonder if the other style - the one where multiple components in different units can be combined, separated by a space - wouldn't be easier to parse AND more human readable. wdyt ?
